src/qibo/quantum_info/quantum_networks.py

```diff
@@ -44,161 +45,204 @@ class QuantumNetwork:
+                + f"Cannot reshape matrix of size {operator.shape} to partition {partition}",
+            )
 
-    def is_hermitian(
-        self, order: Optional[Union[int, str]] = None, precision_tol: float = 1e-8
+    @classmethod
+    def from_operator(
+        cls,
+        arr: np.ndarray,
+        partition: Optional[Union[List[int], Tuple[int]]] = None,
+        system_input: Optional[Union[List[bool], Tuple[bool]]] = None,
+        pure: bool = False,
+        backend=None,
     ):
-        """Returns bool indicating if the Choi operator :math:`\\mathcal{E}` of the network is Hermitian.
+        """Construct a :class:`qibo.quantum_info.QuantumNetwork` object from a ndarray.
 
-        Hermicity is calculated as distance between :math:`\\mathcal{E}` and
-        :math:`\\mathcal{E}^{\\dagger}` with respect to a given norm.
-        Default is the ``Hilbert-Schmidt`` norm (also known as ``Frobenius`` norm).
-
-        For specifications on the other  possible values of the
-        parameter ``order`` for the ``tensorflow`` backend, please refer to
-        `tensorflow.norm <https://www.tensorflow.org/api_docs/python/tf/norm>`_.
-        For all other backends, please refer to
-        `numpy.linalg.norm <https://numpy.org/doc/stable/reference/generated/numpy.linalg.norm.html>`_.
+        This method converts a Choi operator to the internal representation of
+        :class:`qibo.quantum_info.quantum_networks.QuantumNetwork`.
+        The input array can be a pure state, a Choi operator, a unitary operator, etc.
 
         Args:
-            order (str or int, optional): order of the norm. Defaults to ``None``.
-            precision_tol (float, optional): threshold :math:`\\epsilon` that defines if
-                Choi operator of the network is :math:`\\epsilon`-close to Hermicity in
-                the norm given by ``order``. Defaults to :math:`10^{-8}`.
+            arr (ndarray): input numpy array.
+            partition (List[int] or Tuple[int], optional): partition of ``arr``. If ``None``,
+                defaults to the shape of ``arr``. Defaults to ``None``.
+            system_input (List[bool] or Tuple[bool], optional): mask on the input system of the
+                Choi operator. If ``None``, defaults to
+                ``(True,False,True,False...)``, where ``len(system_input)=len(partition)``.
+                Defaults to ``None``.
+            pure (bool, optional): ``True`` when ``arr`` is a "pure" representation (e.g. a pure
+                state, a unitary operator, etc.), ``False`` otherwise. Defaults to ``False``.
+            backend (:class:`qibo.backends.abstract.Backend`, optional): Backend to be used in
+                calculations. If ``None``, defaults to :class:`qibo.backends.GlobalBackend`.
+                Defaults to ``None``.
 
         Returns:
-            bool: Hermiticity condition.
+            :class:`qibo.quantum_info.quantum_networks.QuantumNetwork`:
```

Comment:
The `Returns` item of the docstring needs also a small sentence describing what is being returned.

Reply:
@Canoming also not resolved